### PR TITLE
Fix bfloat16 placement issue in tensorflow abs

### PR DESCRIPTION
**Pull request by Dsantra92**

### CODE REVIEW

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Not sure if the CI run being broken is being caused by the commits in this PR😁
Try to update this branch with master so the CI should run perfectly🙂
Let's see what @VedPatwardhan thinks of this

File: ivy/functional/backends/tensorflow/searching.py [hunk 1]
```diff
@@ -111,7 +111,11 @@ def where(
     out: Optional[Union[tf.Tensor, tf.Variable]] = None,
 ) -> Union[tf.Tensor, tf.Variable]:
     x1, x2 = ivy.promote_types_of_inputs(x1, x2)
-    return tf.cast(tf.experimental.numpy.where(condition, x1, x2), x1.dtype)
+    dev = ivy.dev(x1, as_native=False)
+    ret = tf.cast(tf.experimental.numpy.where(condition, x1, x2), x1.dtype)
+    with tf.device(dev):
```
Comment (vedpatwardhan) [inline ivy/functional/backends/tensorflow/searching.py:116]: Would we need to have this kind of logic for all functions? Probably cleaner to just use the device-shifting decorator in [this](https://github.com/unifyai/ivy/pull/16101) PR

**Review by vedpatwardhan — COMMENTED** [review]
Just a doubt out of curiosity, apart from that I'll just request @sherry30's review as it's device-related. Thanks @Dsantra92 and @hirwa-nshuti 🙂

File: ivy/functional/backends/tensorflow/searching.py (hunk 1 above)
Comment (Dsantra92) [reply ivy/functional/backends/tensorflow/searching.py:116]: Yes, definitely a better way of handling it. Will wait for it to merge.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Review by sherry30 — COMMENTED** [review]
After the merge of `handle_device_shifting` decorator, this seems to work fine because the function gets called in the `tf.device` context manager now.

**Pull request closed without merging**